src/da/da-helper.js

```diff
@@ -0,0 +1,560 @@
+      excludePatterns: ['node_modules', '.git'],
+      verbose: false, // Less verbose for individual page uploads
+      baseFolder: downloadFolder, // preserve shadow folder structure
+    });
+    console.log(chalkDep.green(`Successfully uploaded assets for page ${pageIndex} to ${shadowFolderPath}`));
+  } catch (uploadError) {
+    console.error(chalkDep.red(`Error uploading assets for page ${pageIndex}:`, uploadError.message));
+    throw uploadError;
+  }
+}
+
+/**
+ * Upload an HTML page to DA
+ * @param {string} pageDir - Directory containing the HTML page
+ * @param {string} daLocation - DA location URL
+ * @param {string} token - Authentication token
+ * @param {Object} uploadOptions - Upload options
+ * @param {string} htmlFolder - Base HTML folder for relative path calculation
+ * @param {number} pageIndex - Page index for logging
+ * @param {Object} dependencies - Dependencies for testing (optional)
+ * @return {Promise<void>}
+ */
+async function uploadHTMLPage(pageDir, daLocation, token, uploadOptions, htmlFolder, pageIndex, dependencies = defaultDependencies) {
+  const { chalk: chalkDep, uploadFolder: uploadFolderDep } = dependencies;
+  
+  console.log(chalkDep.yellow(`Uploading updated HTML page ${pageIndex}...`));
+  try {
+    await uploadFolderDep(pageDir, daLocation, token, {
+      ...uploadOptions,
+      fileExtensions: ['.html'],
+      verbose: false,
+      baseFolder: htmlFolder,
+    });
+    console.log(chalkDep.green(`Successfully uploaded HTML page ${pageIndex}`));
+  } catch (uploadError) {
+    console.error(chalkDep.red(`Error uploading HTML page ${pageIndex}:`, uploadError.message));
+    throw uploadError;
+  }
+}
+
+/**
+ * Clean up downloaded assets for a page to free disk space
+ * @param {string} shadowFolderPath - Path to the shadow folder to clean up
+ * @param {number} pageIndex - Page index for logging
+ * @param {Object} dependencies - Dependencies for testing (optional)
+ * @return {Promise<void>}
+ */
+async function cleanupPageAssets(shadowFolderPath, pageIndex, dependencies = defaultDependencies) {
+  const { fs: fsDep, path: pathDep, chalk: chalkDep } = dependencies;
+  
+  console.log(chalkDep.gray(`Cleaning up downloaded assets for page ${pageIndex}...`));
+  try {
+    // Remove the entire shadow folder for this page
+    if (fsDep.existsSync(shadowFolderPath)) {
+      const files = fsDep.readdirSync(shadowFolderPath);
+      for (const file of files) {
+        const filePath = pathDep.join(shadowFolderPath, file);
+        const stat = fsDep.statSync(filePath);
+        if (stat.isFile()) {
+          fsDep.unlinkSync(filePath);
+        }
+      }
+      fsDep.rmdirSync(shadowFolderPath);
+      console.log(chalkDep.gray(`Cleaned up assets for page ${pageIndex}`));
+    }
+  } catch (cleanupError) {
+    console.warn(chalkDep.yellow(`Warning: Could not clean up assets for page ${pageIndex}:`, cleanupError.message));
+  }
```

Comment:
This can be simplified:
```suggestion
  try {
    if (fsDep.existsSync(shadowFolderPath)) {
      fsDep.rmdir(shadowFolderPath, { recursive: true }, (err) => {
        if (err) throw err;
        console.log(chalkDep.gray(`Cleaned up assets for page ${pageIndex}`));
      });
    }
  } catch (cleanupError) {
    console.warn(chalkDep.yellow(`Warning: Could not clean up assets for page ${pageIndex}:`, cleanupError.message));
  }
```